packages/core/src/test/codec.test.ts

```diff
@@ -0,0 +1,72 @@
+import {
+	CodecOrUndefined,
+	getFinalOutputCodec,
+	getOutputCodecOrUndefined,
+	setCodec,
+	setOutputFormat,
+} from '../config/codec';
+
+// getFinalOutputCodec
+
+test('Codec tests valid codec input', () => {
+	const values: CodecOrUndefined[] = ['h264', 'h265', 'vp8', 'vp9'];
+	values.forEach((entry) =>
+		expect(
+			getFinalOutputCodec({codec: entry, emitWarning: true, fileExtension: ''})
```

Comment:
Please set `emitWarning` to `false` everywhere, otherwise it will log noise during tests.